test/functional/tests/performance/test_performance_basic.py

```diff
@@ -0,0 +1,275 @@
+
+    with TestRun.step("Ensure that I/O scheduler for CAS device is 'none'"):
+        TestRun.executor.run_expect_success("sudo echo none > /sys/block/cas1-1/queue/scheduler")
+
+    with TestRun.step("Fill the cache with data via CAS device"):
+        fill_cas_cache(core, cache_line_size)
+        casadm.reset_counters(1, 1)
+
+    with TestRun.step("Measure read performance (throughput and latency) on CAS device"):
+        fio_command.target(core)
+        cas_results = {}
+
+        for nj in num_jobs:
+            fio_command.num_jobs(nj)
+            cas_results[nj] = fio_command.run().pop()
+            TestRun.LOGGER.info(str(cas_results[nj]))
+
+    with TestRun.step("Check if read hit percentage during fio is greater or equal to 99"):
+        cache_stats = cache.get_statistics()
+        read_hits = cache_stats.request_stats.read.hits
+        read_total = cache_stats.request_stats.read.total
+        read_hits_percentage = read_hits / read_total * 100
+        if read_hits_percentage <= 99:
+            TestRun.LOGGER.error(
+                f"Read hits percentage too low: {read_hits_percentage}%\n"
+                f"Read hits: {read_hits}, read total: {read_total}"
+            )
+
+    with TestRun.step("Compare fio results"):
+        for nj in num_jobs:
+            cas_latency = cas_results[nj].read_completion_latency_average().microseconds
+            raw_disk_latency = raw_disk_results[nj].read_completion_latency_average().microseconds
+            cas_read_iops = cas_results[nj].read_iops()
+            raw_disk_iops = raw_disk_results[nj].read_iops()
+            read_iops_ratio = (100 * cas_read_iops) / raw_disk_iops
+
+            TestRun.LOGGER.info(
+                f"Results for num_jobs={nj}, queue_depth={queue_depth},"
+                f" block_size={block_size}, cache_line_size={cache_line_size}"
+            )
+            TestRun.LOGGER.info("Average read latency (us):")
+            TestRun.LOGGER.info(f" - (raw disk) {raw_disk_latency}")
+            TestRun.LOGGER.info(f" - (CAS device) {cas_latency}")
+            TestRun.LOGGER.info("Average read throughput (IOPS):")
+            TestRun.LOGGER.info(f" - (raw disk) {raw_disk_iops}")
+            TestRun.LOGGER.info(f" - (CAS device) {cas_read_iops}")
+            TestRun.LOGGER.info(f"Read ratio: {read_iops_ratio}")
+
+            if read_iops_ratio < 85:
+                TestRun.LOGGER.error(f"The read iops ratio is below expected threshold (85%).")
+
+
+# TODO: for disks other than Intel Optane, fio ramp is needed before fio tests on raw disk
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+@pytest.mark.parametrizex("block_size", [Size(1, Unit.Blocks4096), Size(8, Unit.Blocks4096)])
+@pytest.mark.parametrizex("queue_depth", [1, 16, 32])
+@pytest.mark.parametrizex("cache_line_size", CacheLineSize)
+def test_performance_read_hit_wb(cache_line_size, block_size, queue_depth):
```

Comment:
test names are duplicated - this way they're shadowing each other and the earlier one won't run (I think)

Reply:
They are similar but not duplicated ;)
There is 'w**t**' and 'w**b**' at the and of the name to reflect Write-Through and Writer-Back modes used in test.